tests/python/relax/frontend/test_onnx_frontend.py

```diff
@@ -0,0 +1,1378 @@
+            "b": np.random.normal(size=(48, 64)).astype("float32"),
+        }
+    check_correctness(model, inputs)
+
+
+def test_concat():
+    concat_node = helper.make_node("Concat", ["a", "b"], ["ab"], axis=0)
+
+    graph = helper.make_graph(
+        [concat_node],
+        "concat_test",
+        inputs=[
+            helper.make_tensor_value_info("a", TensorProto.FLOAT, [1, 32]),
+            helper.make_tensor_value_info("b", TensorProto.FLOAT, [1, 32]),
+        ],
+        outputs=[helper.make_tensor_value_info("ab", TensorProto.FLOAT, [2, 32])],
+    )
+
+    model = helper.make_model(graph, producer_name="concat_test")
+    check_correctness(model)
+
+
+def test_add():
+    add_node = helper.make_node("Add", ["a", "b"], ["ab"])
+
+    graph = helper.make_graph(
+        [add_node],
+        "add_test",
+        inputs=[
+            helper.make_tensor_value_info("a", TensorProto.FLOAT, [1, 32]),
+            helper.make_tensor_value_info("b", TensorProto.FLOAT, [1, 32]),
+        ],
+        outputs=[helper.make_tensor_value_info("ab", TensorProto.FLOAT, [1, 32])],
+    )
+
+    model = helper.make_model(graph, producer_name="add_test")
+    check_correctness(model)
+
+
+def test_mul():
+    mul_node = helper.make_node("Mul", ["a", "b"], ["ab"])
+
+    graph = helper.make_graph(
+        [mul_node],
+        "mul_test",
+        inputs=[
+            helper.make_tensor_value_info("a", TensorProto.FLOAT, [1, 32]),
+            helper.make_tensor_value_info("b", TensorProto.FLOAT, [1, 32]),
+        ],
+        outputs=[helper.make_tensor_value_info("ab", TensorProto.FLOAT, [1, 32])],
+    )
+
+    model = helper.make_model(graph, producer_name="mul_test")
+    check_correctness(model)
+
+
+def test_cast():
+    cast_node = helper.make_node("Cast", ["a"], ["a_float"], to=TensorProto.FLOAT)
+
+    graph = helper.make_graph(
+        [cast_node],
+        "cast_test",
+        inputs=[
+            helper.make_tensor_value_info("a", TensorProto.INT32, [1, 32]),
+        ],
+        outputs=[helper.make_tensor_value_info("a_float", TensorProto.FLOAT, [1, 32])],
+    )
+
+    model = helper.make_model(graph, producer_name="cast_test")
+    check_correctness(model)
+
+
+def test_gather():
+    gather_node = helper.make_node("Gather", ["data", "indices"], ["y"], axis=0)
+
+    graph = helper.make_graph(
+        [gather_node],
+        "gather_test",
+        inputs=[
+            helper.make_tensor_value_info("data", TensorProto.FLOAT, [5, 4, 3, 2]),
+            helper.make_tensor_value_info("indices", TensorProto.INT32, [3]),
+        ],
+        outputs=[helper.make_tensor_value_info("y", TensorProto.FLOAT, [3, 4, 3, 2])],
+    )
+
+    model = helper.make_model(graph, producer_name="gather_test")
+    input_values = {
+        "data": np.random.randn(5, 4, 3, 2).astype("float32"),
+        "indices": np.array([0, 1, 3]).astype("int32"),
+    }
+    check_correctness(model, inputs=input_values)
+
+
+def test_gemm():
```

Comment:
GEMM is one of the ops that has various branches for the code generation (e.g., `alpha` being `None`), so I think those branches should be included in the test

Reply:
I updated the test